src/general/OptionsMenu.cs

```diff
@@ -351,6 +358,9 @@ public override void _Ready()
         // We're only utilizing the AcceptDialog's auto resize functionality,
         // so hide the default Ok button since it's not needed
         backConfirmationBox.GetOk().Hide();
+
+        cloudResolutionTitle.RegisterToolTipForControl("cloudResolution", "options", toolTipCallbacks);
+        guiLightEffectsToggle.RegisterToolTipForControl("guiLightEffects", "options", toolTipCallbacks);
```

Comment:
When the data is not needed, would it be nice if it could be just given as null?

Reply:
Could be, assuming the programmer has alternative way for the callbacks handling, otherwise the tooltip wouldn't react to anything, but that could be documented in the code.

Reply:
Well for these tooltips, doesn't the tooltip manager or whatever it is, handle the showing? So the client code that specifies that this should have a tooltip, shouldn't need to worry itself about allocating a dummy callback data.

Reply:
Yes the tooltip manager handles that but it's notified by the callbacks. Now that you mention it, I guess all these callback data can just be stored in the tooltip manager class instead of in the client code.

Reply:
Okay, I ended up unifying all the tooltip callback storing into ToolTipHelper, this should make the custom tooltip more robust and less of a hassle to setup.